### PR TITLE
[WIP] linux4.19: Redo ppc64-dotconfig to boot on G5 [ci skip]

### PR DESCRIPTION
**Pull request by stenstorp**
@q66 Redid the ppc64-dotconfig based on the g5_defconfig. It probably has to be edited to work properly with the LE machines and your build server.

[ci skip]

### CODE REVIEW

**Comment by maxice8** [comment]
add [ci skip] please.

File: srcpkgs/linux4.19/files/ppc64-dotconfig [hunk 1]
```diff
@@ -70,17 +67,17 @@ CONFIG_NO_HZ_IDLE=y
 # CONFIG_NO_HZ is not set
 CONFIG_HIGH_RES_TIMERS=y
 # CONFIG_PREEMPT_NONE is not set
-CONFIG_PREEMPT_VOLUNTARY=y
-# CONFIG_PREEMPT is not set
+# CONFIG_PREEMPT_VOLUNTARY is not set
+CONFIG_PREEMPT=y
```
Comment (q66) [inline srcpkgs/linux4.19/files/ppc64-dotconfig]: Preempt kernel is a no at least for openpower hardware (power8+), please use `PREEMPT_VOLUNTARY` like before

File: srcpkgs/linux4.19/files/ppc64-dotconfig [hunk 2]
```diff
@@ -231,7 +224,7 @@ CONFIG_GENERIC_CPU=y
 CONFIG_PPC_BOOK3S=y
 CONFIG_PPC_FPU=y
 CONFIG_ALTIVEC=y
-CONFIG_VSX=y
+# CONFIG_VSX is not set
```
Comment (q66) [inline srcpkgs/linux4.19/files/ppc64-dotconfig]: we probably want VSX on

File: srcpkgs/linux4.19/files/ppc64-dotconfig [hunk 3]
```diff
@@ -240,162 +233,99 @@ CONFIG_ARCH_ENABLE_HUGEPAGE_MIGRATION=y
 CONFIG_PPC_MM_SLICES=y
 CONFIG_PPC_HAVE_PMU_SUPPORT=y
 CONFIG_PPC_PERF_CTRS=y
-CONFIG_FORCE_SMP=y
 CONFIG_SMP=y
-CONFIG_NR_CPUS=2048
-CONFIG_PPC_DOORBELL=y
+CONFIG_NR_CPUS=128
```
Comment (q66) [inline srcpkgs/linux4.19/files/ppc64-dotconfig]: keep this at 2048

File: srcpkgs/linux4.19/files/ppc64-dotconfig [hunk 4]
```diff
@@ -240,162 +233,99 @@ CONFIG_ARCH_ENABLE_HUGEPAGE_MIGRATION=y
...
 CONFIG_64BIT=y
 CONFIG_MMU=y
-CONFIG_ARCH_MMAP_RND_BITS_MAX=29
-CONFIG_ARCH_MMAP_RND_BITS_MIN=14
-CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MAX=13
-CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MIN=7
+CONFIG_ARCH_MMAP_RND_BITS_MAX=33
+CONFIG_ARCH_MMAP_RND_BITS_MIN=18
+CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MAX=17
+CONFIG_ARCH_MMAP_RND_COMPAT_BITS_MIN=11
```
Comment (q66) [inline srcpkgs/linux4.19/files/ppc64-dotconfig]: don't mess with these either

File: srcpkgs/linux4.19/files/ppc64-dotconfig [hunk 5]
```diff
@@ -240,162 +233,99 @@ CONFIG_ARCH_ENABLE_HUGEPAGE_MIGRATION=y
...
 CONFIG_LOCKDEP_SUPPORT=y
 CONFIG_RWSEM_XCHGADD_ALGORITHM=y
+CONFIG_GENERIC_LOCKBREAK=y
 CONFIG_GENERIC_HWEIGHT=y
 CONFIG_ARCH_HAS_DMA_SET_COHERENT_MASK=y
 CONFIG_PPC=y
 CONFIG_PPC_BARRIER_NOSPEC=y
 CONFIG_EARLY_PRINTK=y
-CONFIG_PANIC_TIMEOUT=0
+CONFIG_PANIC_TIMEOUT=180
```
Comment (q66) [inline srcpkgs/linux4.19/files/ppc64-dotconfig]: or this one

File: srcpkgs/linux4.19/files/ppc64-dotconfig [hunk 6]
```diff
@@ -240,162 +233,99 @@ CONFIG_ARCH_ENABLE_HUGEPAGE_MIGRATION=y
...
-CONFIG_PSERIES_ENERGY=m
-CONFIG_SCANLOG=m
-CONFIG_IO_EVENT_IRQ=y
-CONFIG_LPARCFG=y
-CONFIG_PPC_SMLPAR=y
-CONFIG_CMM=m
-CONFIG_HV_PERF_CTRS=y
-CONFIG_IBMVIO=y
-CONFIG_IBMEBUS=y
+# CONFIG_PPC_POWERNV is not set
```
Comment (q66) [inline srcpkgs/linux4.19/files/ppc64-dotconfig]: powernv platform support is necessary for modern hardware

File: srcpkgs/linux4.19/files/ppc64-dotconfig [hunk 7]
```diff
@@ -448,93 +371,97 @@ CONFIG_HZ_250=y
...
 CONFIG_ARCH_SPARSEMEM_ENABLE=y
 CONFIG_ARCH_SPARSEMEM_DEFAULT=y
 CONFIG_SYS_SUPPORTS_HUGETLBFS=y
 CONFIG_ILLEGAL_POINTER_VALUE=0x5deadbeef0000000
 CONFIG_ARCH_MEMORY_PROBE=y
-CONFIG_NODES_SPAN_OTHER_NODES=y
-# CONFIG_PPC_4K_PAGES is not set
-CONFIG_PPC_64K_PAGES=y
+CONFIG_PPC_4K_PAGES=y
+# CONFIG_PPC_64K_PAGES is not set
```
Comment (q66) [inline srcpkgs/linux4.19/files/ppc64-dotconfig]: we want 64k pages support for sure

File: srcpkgs/linux4.19/files/ppc64-dotconfig [hunk 8]
```diff
@@ -448,93 +371,97 @@ CONFIG_HZ_250=y
...
-CONFIG_FA_DUMP=y
+# CONFIG_KEXEC_FILE is not set
+# CONFIG_RELOCATABLE is not set
+# CONFIG_CRASH_DUMP is not set
 CONFIG_IRQ_ALL_CPUS=y
-CONFIG_NUMA=y
-CONFIG_NODES_SHIFT=8
-CONFIG_USE_PERCPU_NUMA_NODE_ID=y
-CONFIG_HAVE_MEMORYLESS_NODES=y
+# CONFIG_NUMA is not set
```
Comment (q66) [inline srcpkgs/linux4.19/files/ppc64-dotconfig]: we want NUMA

**Comment by q66** [comment]
The main problem with me editing in the changes is that if I do there is a chance it will break the older BE hardware again. Which means I'll need you to reduce the diff as much as possible by yourself...

and yeah, please add [ci skip], otherwise it will clog Travis for other PRs.

**Comment by stenstorp** [comment]
Sorry this took so long, I've been really busy this past week.
Also, don't be afraid submit some changes, so long as you give me the chance to test them before they're committed.

**Comment by q66** [comment]
You know, just adjusting these things I pointed out will not change anything, since changing an option in Linux kernel config will also affect a load of other options, so *do not* ever manually change kernel config options, always do it via menuconfig. As I said previously, try to minimize the diff. When I was working on mine, I did something like...

1) keep an older reference "working" configuration (this would be your newly generated config)
2) keep a "broken" config close to what I wanted to achieve (that would be my current ppc64-defconfig)
3) then menuconfig a bunch of changes, bringing the diff closer to the "broken" config
4) build, test, if it works, replace the reference configuration in 1) with the newly updated configuration
5) repeat 3) until the config no longer works, then find out what the breakage was

**Comment by q66** [comment]
Examples of how manually messing with config will leave it inconsistent: `CONFIG_PREEMPT_VOLUNTARY` implies `CONFIG_TREE_RCU` rather than `CONFIG_PREEMPT_RCU`; also enabling `CONFIG_PPC_POWERNV` will also automatically enable a load of other options that were previously not there because they were made invisible (as they're dependent on it), etc.

**Comment by q66** [comment]
Anyway, I plan to diff our BE config against Debian's and see if there's anything obvious or anything we should correct anyway, then post a pull request and I'll need you to test it.

**Comment by q66** [comment]
Okay, I looked over the diff, and there isn't really anything much I would like to change. Here are some things you can enable while fixing it, though, as I'd like them enabled:

```
CONFIG_PSERIES_ENERGY=y
CONFIG_PS3_ADVANCED=y
CONFIG_PS3_VRAM=m
CONFIG_ADB_PMU=y
CONFIG_ADB_PMU_LED=y
# CONFIG_ADB_PMU_LED_DISK is not set
CONFIG_PMAC_SMU=y
CONFIG_MAC_EMUMOUSEBTN=y
CONFIG_WINDFARM_PM81=m
CONFIG_WINDFARM_PM72=m
CONFIG_WINDFARM_RM31=m
CONFIG_WINDFARM_PM91=m
CONFIG_WINDFARM_PM112=m
CONFIG_WINDFARM_PM121=m
CONFIG_PMAC_RACKMETER=m
CONFIG_ARCH_WANTS_FREEZER_CONTROL=y
CONFIG_HIBERNATE_CALLBACKS=y
CONFIG_HIBERNATION=y
CONFIG_PM_STD_PARTITION=""
CONFIG_PCIEPORTBUS=y
CONFIG_HOTPLUG_PCI_PCIE=y
CONFIG_PCIEAER=y
CONFIG_PCIEAER_INJECT=m
# CONFIG_PCIE_ECRC is not set
CONFIG_PCIEASPM=y
# CONFIG_PCIEASPM_DEBUG is not set
CONFIG_PCIEASPM_DEFAULT=y
# CONFIG_PCIEASPM_POWERSAVE is not set
# CONFIG_PCIEASPM_POWER_SUPERSAVE is not set
# CONFIG_PCIEASPM_PERFORMANCE is not set
CONFIG_PCIE_PME=y
CONFIG_PCIE_DPC=y
CONFIG_PCIE_PTM=y
```

maybe one of them might fix the thing, too; some of these enable things necessary for Power Macs (`CONFIG_PMAC_SMU` particularly)

please menuconfig these options in until the .config contains those entries exactly as above, then re-test.

**Comment by stenstorp** [comment]
I've done a bit of messing about with the old ppc64 config (the one currently in the repo) and with minimal tweaking, got it to boot. I had to disable the relocatable kernel and crash_dump. There seem to be some issues with NFS which cause mkinitrd segfault, messing up the initramfs.
Once I find out whats failing, I'll submit the new config with a smaller diff this time :)

**Comment by q66** [comment]
Great. Please incorporate the changes above also. They are mostly specific to big endian machines in practice so it should expand our hw support.

**Comment by maxice8** [comment]
ping

**Comment by stenstorp** [comment]
Been busy (started back at university) barely looked at void. Got a bit of spare time so I'll be getting on with this soon.

**Comment by stenstorp** [comment]
Closing since @pullmoll has a better kernel config.

**Pull request closed without merging**